pkg/api/v1alpha1/tinkerbellmachineconfig.go

```diff
@@ -27,6 +27,7 @@ func NewTinkerbellMachineConfigGenerate(name string, opts ...TinkerbellMachineCo
 		Spec: TinkerbellMachineConfigSpec{
 			HardwareSelector: HardwareSelector{},
 			OSFamily:         Ubuntu,
+			OSImageURL:       "",
```

Comment:
any reason why we are exposing this in the machine config instead of datacenter config? I would think most users would use OSImageURL there generally

Reply:
So basically when we defaulted to ubuntu we missed adding OSImageURL field. For modular upgrades machine configs is expected in OSImageURL field. Also, this is more akin to how we set templated at machine config level in other providers. Once we introduced OSImageURL on machine config level the plan was to give preference to this field and eventually deprecate OSImageURL field at Datacenter config. But we haven't reached there yet!